src/bluesearch/entrypoint/database/add.py

```diff
@@ -0,0 +1,87 @@
+"""Adding an article to the database."""
+import argparse
+import json
+
+import sqlalchemy
+
+import bluesearch.database.article as article_module
+
+
+def get_parser() -> argparse.ArgumentParser:
+    """Create a parser."""
+    parser = argparse.ArgumentParser(
+        description="Add entries.",
+    )
+    parser.add_argument(
+        "db_url",
+        type=str,
+        help="""
+        The location of the database depending on the database type.
+
+        For MySQL the server URL should be provided, for SQLite the
+        location of the database file. Generally, the scheme part of
+        the URL should be omitted, e.g. for MySQL the URL should be
+        of the form 'my_sql_server.ch:1234/my_database' and for SQLite
+        of the form '/path/to/the/local/database.db'.
+        """,
+    )
+    parser.add_argument(
+        "parser",
+        type=str,
+        help="""Parser class.""",
+    )
+    parser.add_argument(
+        "path",
+        type=str,
+        help="""Path to the file/directory to be parsed.""",
+    )
+    parser.add_argument(
+        "--db-type",
+        default="sqlite",
+        type=str,
+        choices=("mysql", "sqlite"),
+        help="Type of the database.",
+    )
+    return parser
+
+
+def run(
+    *,
+    db_url: str,
+    parser: str,
+    path: str,
+    db_type: str,
+) -> None:
+    """Add an entry to the database.
+
+    Parameter description and potential defaults are documented inside of the
+    `get_parser` function.
+    """
+    if db_type == "sqlite":
+        engine = sqlalchemy.create_engine(f"sqlite:///{db_url}")
+
+    elif db_type == "mysql":
+        raise NotImplementedError
+
+    else:
+        raise ValueError(f"Unrecognize database type {db_type}.")  # pragma: nocover
+
+    valid_parsers = [x.__name__ for x in article_module.ArticleParser.__subclasses__()]
```

Comment:
Just a small caveat here: if a subclass is in a different module, then this module needs to be imported first, otherwise the subclass won't be listed. At the moment we have all subclasses in the same module, but it's possible we'll refactor in the future, and it's something easy to miss.

Without trying to start an argument, just wanted to mention again that I'm not a super big fan of dynamic name resolutions and turning strings into python code. So everytime I see `__import__`, `getattr`, `eval`, `vars(...)`, `func(**kwargs)`, etc. a part of me gets a bit sad. I think this makes the code less readable and harder to reason about, probably also stops IDEs/mypy from properly resolving objects too. It's a bit personal though, so no offence!

I just feel that if there are O(1) possible options like in the present case, then they can be easily enumerated/listed. Personally I would probably try and use dictionaries / enums / some kind of simple registry.

Anyway, no offence, just contributing my 2 cents of philosophical thought, we can carry on 😄

Reply:
I am totally fine with having an if statement:)

Reply:
Implemented in 89733d4bdfc57502e3e738dd7babce18209817f6

Mental note for all of us. Whenever we implement a new parser, we need to also modify the `parse` entrypoint.